nixos/modules/services/misc/gitea.nix

```diff
@@ -400,12 +413,31 @@ in
         message = ''
           When creating a database via NixOS, the db user and db name must be equal!
           If you already have an existing DB+user and this assertion is new, you can safely set
-          `services.gitea.createDatabase` to `false` because removal of `ensureUsers`
+          `services.gitea.database.createDatabase` to `false` because removal of `ensureUsers`
           and `ensureDatabases` doesn't have any effect.
         '';
       }
     ];
 
+    services.gitea.database = lib.mkMerge [
+      (lib.mkIf cfg.database.createDatabase {
+        host = lib.mkDefault "127.0.0.1";
+        port = lib.mkDefault (if usePostgresql then pg.settings.port else 3306);
+        name = lib.mkDefault "gitea";
+        user = lib.mkDefault "gitea";
+        socket = lib.mkDefault (if usePostgresql then "/run/postgresql" else if useMysql then "/run/mysqld/mysqld.sock" else null);
+        path = lib.mkDefault "${cfg.stateDir}/data/gitea.db";
+      })
+      {
+        # Create database passwordFile default when password is configured.
+        passwordFile =
+          mkDefault (toString (pkgs.writeTextFile {
```

Comment:
I don't think we could use mkDefault here even if it is breaking to make people notice when they have different passwords configured